sympy/geometry/shape.py

```diff
@@ -0,0 +1,318 @@
+""" 2D geometrical entity constructed by boolean operations
+    of various geometric elements.
+
+"""
+
+from __future__ import division, print_function
+from sympy.core import S, pi, sympify
+from sympy.core.compatibility import ordered
+from sympy.core.symbol import _symbol
+from sympy.core.numbers import Rational, oo
+from sympy import symbols, simplify, solve
+from sympy.geometry.entity import GeometryEntity, GeometrySet
+from sympy.geometry.point import Point, Point2D
+from sympy.geometry.line import Line, Line2D, Ray2D, Segment2D, LinearEntity3D
+from sympy.geometry.ellipse import Ellipse, Circle
+from sympy.geometry.parabola import Parabola
+from sympy.functions.elementary.trigonometric import asin
+
+from sympy import Abs, sqrt
+
+class Shape(GeometrySet):
+    """
+    Parameters
```

Comment:
Please add a description to the top of the docstring describing what the class is.

Reply:
okk